devito/mpi/halo_scheme.py

```diff
@@ -121,7 +156,10 @@ def union(self, halo_schemes):
         Create a new HaloScheme from the union of a set of HaloSchemes.
         """
         halo_schemes = [hs for hs in halo_schemes if hs is not None]
-        if not halo_schemes:
+
+        if len(halo_schemes) == 1:
+            return halo_schemes[0]
```

Comment:
why this change?

Reply:
There is zero work done when you have one haloscheme...so early exit...